### PR TITLE
Provision for notes while editing and undoing a transaction

### PR DESCRIPTION
**Pull request by KoustavMuhuri**
Code Changes for providing notes while editing and undoing a transaction. Functionality for Showing Notes on Click of a particular transaction.

### CODE REVIEW

**Comment by ippezshelby** [comment]
KoustavMuhuri, Could you please add the same notes for Savings Transactions - Deposit and Withdrawal + Undo Transaction for Savings Transactions
Issue at Jira is at https://mifosforge.jira.com/browse/MIFOSX-2620

Thanks

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/note/domain/NoteRepository.java [hunk 1]
```diff
@@ -41,5 +41,6 @@
 
     List<Note> findBySavingsAccountId(Long savingAccountId);
 
+    Note findByLoanTransactionId(Long loanTransactionId);
```
Comment (rajuan) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/note/domain/NoteRepository.java:44]: By data design there can be multiple notes for a single transaction id.... this API cannot return single Note

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java [hunk 2]
```diff
@@ -888,20 +888,23 @@ public CommandProcessingResult adjustLoanTransaction(final Long loanId, final Lo
...
+        changes.put("note", noteText);  
+         Note note = null;  
+         /**  
+          * If a new transaction is not created, associate note with the  
+          * transaction to be adjusted  
+          **/  
+         if (newTransactionDetail.isGreaterThanZero(loan.getPrincpal().getCurrency())) {  
+             note = Note.loanTransactionNote(loan, newTransactionDetail, noteText);  
+         } else {  
+            note = this.noteRepository.findByLoanTransactionId(transactionId);
```
Comment (rajuan) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java:900]: This API cannot return a single note object... so premise of this solution becomes invalid.... What is the expectation of this PR? can you please explain or point us to relevant JIRA issue?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java [hunk 3]
```diff
@@ -1286,11 +1303,13 @@ public String accountTransferSchema() {
...
                     + " totran.transaction_date as toTransferDate, totran.amount as toTransferAmount,"
-                    + " totran.description as toTransferDescription "
+                    + " totran.description as toTransferDescription ,"
+                    + " usernote.note as note"
                     + " from m_loan l join m_loan_transaction tr on tr.loan_id = l.id"
                     + " join m_currency rc on rc.`code` = l.currency_code "
                     + " left join m_account_transfer_transaction fromtran on fromtran.from_loan_transaction_id = tr.id "
-                    + " left join m_account_transfer_transaction totran on totran.to_loan_transaction_id = tr.id ";
+                    + " left join m_account_transfer_transaction totran on totran.to_loan_transaction_id = tr.id "
+                    + "left join m_note usernote on usernote.loan_transaction_id=tr.id";
```
Comment (rajuan) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java:1312]: What is the use of this join when the attribute is not read in mapper?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java [hunk 4]
```diff
@@ -516,13 +522,24 @@ public LoanTransactionData retrieveLoanTransaction(final Long loanId, final Long
 
         final LoanTransaction transaction = this.loanTransactionRepository.findOne(transactionId);
         if (transaction == null) { throw new LoanTransactionNotFoundException(transactionId); }
-
+        Note note = this.noteRepository.findByLoanTransactionId(transactionId);
```
Comment (rajuan) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java:525]: By data design note cannot be unique

**Comment by rajuan** [comment]
JIRA issue not mentioned, I couldn't understand the requirement for this PR. May be you can make use of Notes APIs. Please discuss your requirement and solution approach in dev forum for clarity and help.

**Comment by KoustavMuhuri** [comment]
Sir,Thanks for your observations. I have noted down the observations.

**Pull request closed without merging**